### PR TITLE
[Table Model] Support basic table management

### PR DESCRIPTION
**Pull request by MarcosZyk**
### Supported Features

1. create table
2. describe table
3. show tables

### Result

```
IoTDB> create database db_1;
Msg: The statement is executed successfully.
IoTDB> create table db_1.t_1(device TEXT ID, attr TEXT ATTRIBUTE, s INT32 MEASUREMENT);
Msg: The statement is executed successfully.
IoTDB> use db_1
Msg: The statement is executed successfully.
IoTDB> create table t_2(region TEXT ID, code TEXT ID, type TEXT ATTRIBUTE, price TEXT ATTRIBUTE, s_1 INT32 MEASUREMENT, s_2 FLOAT MEASUREMENT);
Msg: The statement is executed successfully.
IoTDB> show tables
+---------+----+
|TableName| TTL|
+---------+----+
|      t_2|null|
|      t_1|null|
+---------+----+
Total line number = 2
It costs 0.043s
IoTDB> describe t_2
+----------+--------+-----------+
|ColumnName|DataType|   Category|
+----------+--------+-----------+
|      Time|   INT64|       TIME|
|    region|    TEXT|         ID|
|      code|    TEXT|         ID|
|      type|    TEXT|  ATTRIBUTE|
|     price|    TEXT|  ATTRIBUTE|
|       s_1|   INT32|MEASUREMENT|
|       s_2|   FLOAT|MEASUREMENT|
+----------+--------+-----------+
Total line number = 7
It costs 0.008s
IoTDB> describe db_1.t_1
+----------+--------+-----------+
|ColumnName|DataType|   Category|
+----------+--------+-----------+
|      Time|   INT64|       TIME|
|    device|    TEXT|         ID|
|      attr|    TEXT|  ATTRIBUTE|
|         s|   INT32|MEASUREMENT|
+----------+--------+-----------+
Total line number = 4
It costs 0.005s
IoTDB> create table if not exists t_2(region TEXT ID, code TEXT ID, type TEXT ATTRIBUTE, price TEXT ATTRIBUTE, s_1 INT32 MEASUREMENT, s_2 FLOAT MEASUREMENT);
Msg: The statement is executed successfully.

```

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnSchema.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Objects;
+
+public abstract class ColumnSchema {
```
Comment (JackieTien97) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnSchema.java:30]: rename this class? we've already got a ColumnSchema.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnCategory.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+package org.apache.iotdb.commons.schema.table.column;
+
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public enum ColumnCategory {
```
Comment (JackieTien97) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnCategory.java:29]: We already have this class name.

File: iotdb-protocol/thrift-confignode/src/main/thrift/confignode.thrift [hunk 3]
```diff
@@ -1553,5 +1555,11 @@ service IConfigNodeRPCService {
 
   /** Get throttle quota information */
   TThrottleQuotaResp getThrottleQuota()
+
+  // ======================================================
+  // Table
+  // ======================================================
+
+  common.TSStatus createTable(binary tableInfo)
```
Comment (JackieTien97) [inline iotdb-protocol/thrift-confignode/src/main/thrift/confignode.thrift:1563]: is this interface used for write auto create? If so, just updating total tabel info each time will have problems in concurrent case?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-protocol/thrift-confignode/src/main/thrift/confignode.thrift (hunk 3 above)
Comment (MarcosZyk) [reply iotdb-protocol/thrift-confignode/src/main/thrift/confignode.thrift:1563]: No. This is just the interface for executing manual table creation.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/state/schema/CreateTableState.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.confignode.procedure.state.schema;
+
+public enum CreateTableState {
+  PRE_CREATE,
+  PRE_RELEASE,
+  VALIDATE_TIMESERIES_EXISTENCE,
```
[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/state/schema/CreateTableState.java:25]: This is used for preventing timeseries creation under given table node via Tree model and table creation overlapping with existing MTree path.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/schema/ConfigMTree.java [hunk 5]
```diff
@@ -621,6 +628,81 @@ private IConfigMNode getNodeSetTemplate(int templateId, PartialPath path)
 
   // endregion
 
+  // region table management
+
+  public synchronized void preCreateTable(PartialPath database, TsTable table)
```
[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/schema/ConfigMTree.java]: Removed.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/schema/ClusterSchemaInfo.java [hunk 6]
```diff
@@ -1039,6 +1044,80 @@ public Map<String, TDatabaseSchema> getMatchedDatabaseSchemasByOneName(
...
+    } catch (MetadataException e) {
+      return RpcUtils.getStatus(e.getErrorCode(), e.getMessage());
+    } finally {
+      databaseReadWriteLock.writeLock().unlock();
+    }
+  }
+
+  public void getTable() {}
+
+  public void getAllTables() {}
```
[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/schema/ClusterSchemaInfo.java]: Removed.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnSchema.java (hunk 1 above)
Comment (MarcosZyk) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnSchema.java]: Renamed to TsTableColumnSchema.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnCategory.java (hunk 2 above)
Comment (MarcosZyk) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/column/ColumnCategory.java]: Renamed to TsTableColumnCategory.

**Pull request merged**